guides/v2.0/extension-dev-guide/code-generation.md

```diff
@@ -26,15 +26,11 @@ Provided the Magento application is not set for <a href="{{ site.gdeurl }}config
 
 In particular,
 
-*	A <a href="{{ site.gdeurl }}extension-dev-guide/depend-inj.html#dep-inj-mod-type-fact">Factory</a> class creates instances of a type. For example, a generated <code>\Magento\Customer\Model\AddressFactory</code> creates new instances of <code>\Magento\Customer\Model\Address</code>. The code in <code>AddressFactory</code> provides "typeSafety" (@return annotation) so IDEs understand the type of returned object for the <code>Address</code> type.
+*	A Factory class creates instances of a type. See <a href="{{ site.gdeurl }}extension-dev-guide/factories.html">Instantiating objects with factories</a> for more information. Factories are directly referenced within application code.
 
+*	You can designate a Proxy to be generated for a type in order to ensure the type is not instantiated until it is needed. See <a href="{{ site.gdeurl }}extension-dev-guide/proxies.html">Proxies</a> for more information. Proxies are directly referenced within application code.
```

Comment:
Whatever is convenient. You can also paste the text here and I'll do it when I merge this PR.